lib/util.js

```diff
@@ -222,6 +267,19 @@ function urlToPath(path) {
     };
 }
 
+function getMinimumMultipartPartSizeForTransfer(options, preferredPartSize){
+    let minMemoryBlockSize = preferredPartSize;
+    if(options && options.uploadFiles && options.uploadFiles.length > 0){
+        let minMultipartPartSize = -1;
+        // find max minPartSize from the uploadFiles array (as it defines the largest minimal memory block)
+        options.uploadFiles.forEach(current => {
+            minMultipartPartSize = Math.max(minMultipartPartSize, current.minPartSize);
+        });
+        minMemoryBlockSize = Math.max(minMemoryBlockSize, minMultipartPartSize);
+    }
+    return minMemoryBlockSize;
+}
```

Comment:
I'm not sure this is required. I'll do a separate review of the memory allocator.

Reply:
Some tests suggested it is - we have tests in which parts have different sizes. It should not happen in normal use (except, maybe, for the last block that could be smaller as the others), but nothing prevents it from happening either. Since I base the underlying buffer size used by the memory allocator on that block size, I prefer to be able to have an integer number of blocks being able to fit in the buffer for better concurrency and memory usage (see an example of underlying buffer creation based on block size [here](https://github.com/adobe/node-httptransfer/blob/b4f91c97100322faa6b9b3deac1b899136886b9b/lib/block/blockdownload.js#L171-L173)).